src/ucp/core/ucp_ep.inl

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+
+#ifndef UCP_EP_INL_
+#define UCP_EP_INL_
+
+#include "ucp_ep.h"
+#include "ucp_worker.h"
+
+#include <ucs/arch/bitops.h>
+
+
+static inline ucp_ep_config_t *ucp_ep_config(ucp_ep_h ep)
+{
+    return &ep->worker->ep_config[ep->cfg_index];
+}
+
+static inline ucp_lane_index_t ucp_ep_get_am_lane(ucp_ep_h ep)
+{
+    ucs_assert(ucp_ep_config(ep)->key.am_lane != UCP_NULL_RESOURCE);
+    return ep->am_lane;
+}
+
+static inline ucp_lane_index_t ucp_ep_get_wireup_msg_lane(ucp_ep_h ep)
+{
+    return ucp_ep_config(ep)->key.wireup_msg_lane;
+}
+
+static inline uct_ep_h ucp_ep_get_am_uct_ep(ucp_ep_h ep)
+{
+    return ep->uct_eps[ucp_ep_get_am_lane(ep)];
+}
+
+static inline ucp_rsc_index_t ucp_ep_get_rsc_index(ucp_ep_h ep, ucp_lane_index_t lane)
+{
+    return ucp_ep_config(ep)->key.lanes[lane];
+}
+
+static inline ucp_rsc_index_t ucp_ep_num_lanes(ucp_ep_h ep)
+{
+    return ucp_ep_config(ep)->key.num_lanes;
+}
+
+static inline ucp_rsc_index_t ucp_ep_pd_index(ucp_ep_h ep, ucp_lane_index_t lane)
+{
+    ucp_context_h context = ep->worker->context;
+    return context->tl_rscs[ucp_ep_get_rsc_index(ep, lane)].pd_index;
+}
+
+static inline uct_pd_h ucp_ep_pd(ucp_ep_h ep, ucp_lane_index_t lane)
+{
+    ucp_context_h context = ep->worker->context;
+    return context->pds[ucp_ep_pd_index(ep, lane)];
+}
+
+static inline ucp_pd_map_t ucp_lane_map_get_lane(ucp_pd_lane_map_t lane_map,
+                                                 ucp_lane_index_t lane)
+{
+    return (lane_map >> (lane * UCP_PD_INDEX_BITS)) & UCS_MASK(UCP_PD_INDEX_BITS);
+}
+
+static inline const char* ucp_ep_peer_name(ucp_ep_h ep)
+{
+#if ENABLE_DEBUG_DATA
+    return ep->peer_name;
+#else
+    return "??";
```

Comment:
@yosefe - maybe return something more informative ? No debug enabled ? Or enable debug for more info ?